p2p/address-book/src/book.rs

```diff
@@ -220,25 +206,28 @@ impl<Z: NetworkZone> AddressBook<Z> {
         self.white_list.remove_peers_with_ban_id(&addr.ban_id());
         self.gray_list.remove_peers_with_ban_id(&addr.ban_id());
 
-        self.banned_peers.insert(addr.ban_id());
-        self.banned_peers_fut
-            .push(BanedPeerFut(addr.ban_id(), sleep(time)))
+        let unban_at = Instant::now() + time;
+
+        self.banned_peers_queue.insert_at(addr.ban_id(), unban_at);
+        self.banned_peers.insert(addr.ban_id(), unban_at);
     }
 
     /// adds a peer to the gray list.
     fn add_peer_to_gray_list(&mut self, mut peer: ZoneSpecificPeerListEntryBase<Z::Addr>) {
         if self.white_list.contains_peer(&peer.adr) {
+            tracing::trace!("Peer {} is already in white list skipping.", peer.adr);
```

Comment:
nit: should be `addr` like everywhere else

Reply:
this is how Monero's p2p network encodes the field name in this message, I could change it but I don't think it's too bad